jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/factory/JMSConnectionFactory.java

```diff
@@ -520,7 +520,7 @@ public MessageProducer createMessageProducer(Session session, Destination destin
                 return session.createProducer(destination);
             } else {
                 if (this.destinationType.equals(JMSConstants.JMSDestinationType.QUEUE)) {
-                    return ((QueueSession) session).createProducer((Queue) destination);
+                    return ((QueueSession) session).createSender((Queue) destination);
```

Comment:
This is to support older JMS versions. Earlier versions didn't have a generic createProducer method. They had createSender for queues and createPublisher for topics.